src/api/mod.rs

```diff
@@ -1390,6 +1413,78 @@ impl<'a> AuthenticatedApi<'a> {
     }
 }
 
+/// Available datasets for fetching organization events
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Dataset {
+    /// Our logs dataset
+    OurLogs,
+}
+
+impl Dataset {
+    /// Returns the string representation of the dataset
+    fn as_str(&self) -> &'static str {
+        match self {
+            Dataset::OurLogs => "ourlogs",
+        }
+    }
+}
+
+impl fmt::Display for Dataset {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.as_str())
+    }
+}
+
+/// Options for fetching organization events
+pub struct FetchEventsOptions<'a> {
+    /// Dataset to fetch events from
+    pub dataset: Dataset,
+    /// Fields to include in the response
+    pub fields: &'a [&'a str],
+    /// Project ID to filter events by
+    pub project_id: Option<&'a str>,
```

Comment:
Is this optional as far as the backend is concerned? If yes, let's also make it optional when running the command (currently, it is required to specify a project). If not, let's make the struct field required:


```suggestion
    pub project_id: &'a str,
```

Reply:
FWIW I'm not sure whether users know what a _project_id_ actually is, because this is not the slug, but the numeric identifier. Do we have a facility in sentry-cli to map from a project slug to a project id?

Reply:
+ Agreed to optional, but I think then the output needs to be adapted to include the project slug, so that users can understand which project a specific log entry is from.

Reply:
> Do we have a facility in sentry-cli to map from a project slug to a project id?

No, we want to add this in a future PR though, so that the logs command can also take project slugs (rather would not add it to this PR, as it is already quite big as is)

All other CLI commands support project slugs natively – the backend APIs we call can take either the ID or the slug

Reply:
Let's make this required for now, we can also make the API more lenient later, for now I'd like this first version to go out ASAP